src/common/services/randomNumber.js

```diff
@@ -0,0 +1,40 @@
+import seedrandom from 'seedrandom'
+import window from 'global/window'
+
+export class RandomNumberService {
+  /**
+   * @type {Function?}
+   */
+  seededRandom = null
+
+  constructor() {
+    // The availability of window.location needs to be checked before accessing
+    // its .search property. This code runs in both a browser and Node.js
+    // context, and window.location is not defined in Node.js environments.
+    const initialSeed = new URLSearchParams(window.location?.search).get('seed')
```

Comment:
you don't need to check for the existence of `window` though? that's what i more commonly see 🤔

Reply:
Not with the `global/window` module being imported on line 2. 🙂